### PR TITLE
Vectorize marker of galaxy

### PR DESCRIPTION
**Pull request by 10110111**
### Old

stellarium-009

### New

stellarium-010

### CODE REVIEW

File: src/core/modules/Nebula.cpp [hunk 1]
```diff
@@ -962,12 +994,18 @@ void Nebula::drawHints(StelPainter& sPainter, float maxMagHints, StelCore *core)
...
 		case NebRGx:
-		case NebGxCl:			
-			if (finalSize > 35.f)
-				Nebula::texGalaxyLarge->bind();
-			else
-				Nebula::texGalaxy->bind();
-			break;
+		case NebGxCl:
+		{
+			// The rotation angle in renderEllipticMarker() is relative to screen. Make sure to compute correct angle from 90+orientationAngle.
```
Comment (gzotti) [inline src/core/modules/Nebula.cpp:999]: After moving this block here, you can delete it from below.

Comment (10110111) [reply src/core/modules/Nebula.cpp:999]: Indeed.

**Review by gzotti — APPROVED** [review]
I still don't like "auto", but it's OK for me. Thanks.

**Comment by barbu34** [comment]
Since I installed version 23.2 of stellarium the galaxies are sometimes displayed in white and at other times in white depending on their position on the screen. Ditto for the globular clusters, they go from yellow to white depending on where they are on the screen.
Bug-galaxyGlobAmas-colo2r
Bug-galaxyGlobAmas-color

**Comment by gzotti** [comment]
#3302

**Comment by barbu34** [comment]
2023-08-20T12:40:14
Operating System: Windows 11 Version 22H2
Compiled using MSVC++ 16.10, 16.11 (Visual Studio 2019)
Qt runtime version: 6.4.3
Qt compilation version: 6.4.3
Addressing mode: 64-bit
Total physical memory: 16085 MB
Available physical memory: 11710 MB
Physical memory in use: 27%
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
Processor speed: 2304 MHz
Processor name: 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
C:\Program Files\Stellarium\stellarium.exe --mesa-mode -platform windows:altgr 
 -------------------------------------------------------------
[ This is Stellarium 23.2 (v23.2.0) - https://stellarium.org/ ]
[ Copyright (C) 2000-2023 Stellarium Developers               ]
 ------------------------------------------------------------
Default surface format:  QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::NoProfile)
OpenGL module type: desktop OpenGL
Warning: Setting a new default format with a different version or profile after the global shared context is created may cause issues with context sharing.
Default surface format:  QSurfaceFormat(version 3.3, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
OpenGL module type: desktop OpenGL
StelGLWidget constructor
StelGraphicsScene constructor
initializeGL(windowWidth = 1920, windowHeight = 1080)
OpenGL supported version:  "3.0 Mesa 11.2.2"
Current Format:  QSurfaceFormat(version 3.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize 8, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::DoubleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::NoProfile)
StelMainView::init
Luminance textures are not supported
Anisotropic filtering is not supported!
Maximum 2D texture size: 8192
Detected: OpenGL "3.0"
Driver version string: "3.0 Mesa 11.2.2"
GL vendor is "VMware, Inc."
GL renderer is "Gallium 0.4 on llvmpipe (LLVM 3.6, 256 bits)"
GL Shading Language version is "1.30"
MESA Version Number detected:  11.2
Mesa version is fine, we should not see a graphics problem.
GLSL Version Number detected:  1.3
GLSL version is fine, we should not see a graphics problem.
stel.OpenGLArray: Vertex Array Objects are supported
Sky language is  "fr_FR"
Planetary features language is  "fr_FR"
Application language is  "fr_FR"
Scripts language is  "fr_FR"
Cache directory is:  "C:\\Users\\balto\\AppData\\Local\\stellarium\\stellarium\\cache"
Loaded 252 countries
Loaded 193 regions
Loading Solar System data (1: planets and moons) ...
Loading from : "./data/ssystem_major.ini"
SolarSystem has  83 entries.
Loaded 83 Solar System bodies
Loading Solar System data (2: minor bodies)...
Loading from : "C:\\Users\\balto\\AppData\\Roaming\\Stellarium/data/ssystem_minor.ini"
qt.gui.imageio: libpng warning: iCCP: profile 'icc': 'GRAY': Gray color space not permitted on RGB PNG
SolarSystem has  2248 entries.
Loaded 2165 Solar System bodies
File ssystem_minor.ini is loaded successfully...
qt.gui.imageio: libpng warning: iCCP: profile 'ICC profile': 'RGB ': RGB color space not permitted on grayscale PNG
Loading nomenclature for Solar system bodies ...
Loaded 15924 / 15926 items of planetary surface nomenclature
WARNING - These planets to assign nomenclature items were not found: "Dactyl"
Loading star data ...
"Loading .\\stars\\default\\stars_0_0v0_8.cat: 0_0v0_8; 4979"
"Loading .\\stars\\default\\stars_1_0v0_8.cat: 1_0v0_8; 21806"
"Loading .\\stars\\default\\stars_2_0v0_8.cat: 2_0v0_8; 150826"
"Loading .\\stars\\default\\stars_3_1v0_4.cat: 3_1v0_4; 425807"
"Loading C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\stars\\default\\stars_4_1v0_2.cat: 4_1v0_4; 1692779"
"Loading C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\stars\\default\\stars_5_2v0_1.cat: 5_2v0_1; 7083058"
"Loading C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\stars\\default\\stars_6_2v0_1.cat: 6_2v0_1; 24670782"
"Loading C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\stars\\default\\stars_7_2v0_1.cat: 7_2v0_1; 50733321"
"Loading C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\stars\\default\\stars_8_2v0_1.cat: 8_2v0_1; 92304337"
Finished loading star catalogue data, max_geodesic_level:  8
Loading scientific star names from .\stars\default\name.fab
Loaded 4942 / 4942 scientific star names
Loading scientific star extra names from .\stars\default\extra_name.fab
Loaded 26200 / 26200 scientific star extra names
Loading variable stars from .\stars\default\gcvs_hip_part.dat
Loaded 6862 / 6862 variable stars
Loading double stars from .\stars\default\wds_hip_part.dat
Loaded 22992 / 22992 double stars
Loading cross-identification data from .\stars\default\cross-id.dat
Loaded 108378 / 108378 cross-identification data records for stars
Loading parallax errors data from .\stars\default\hip_plx_err.dat
Loaded 117703 / 117703 parallax error data records for stars
Loading proper motion data from .\stars\default\hip_pm.dat
Loaded 117955 / 117955 proper motion data records for stars
navigation/preset_sky_time is a double - treating as jday: "2451514.25001"
Loading DSO data ...
[...] Stellarium DSO Catalog, version 3.19 (standard edition)
Loaded 94661 DSO records
Loading DSO outline data ...
Loaded 98 DSO outline records successfully
Loading DSO discovery data ...
Loaded 183 / 183 DSO discovery records successfully
LandscapeMgr: initialized Cache for 100 MB.
LandscapeMgr::setCurrentLandscapeID: unknown landscape "Jardin" , using 'zero'
Unable to find module called "GridLinesMgr"
Loading star names from .\skycultures\modern\star_names.fab
Loaded 1104 / 1104 common star names
Loading DSO name data ...
Loaded 1321 / 1346 DSO name records successfully
WARNING - No position data for 25 objects: PGC 2798, PGC 2907, PGC 23521, PGC 28759, PGC 29167, PGC 34658, PGC 42102, PGC 44750, PGC 54559, PGC 59117, PGC 59858, PGC 59953, PGC 69018, PGC 69877, PGC 73957, PGC 95597, PGC 100170, PGC 100170, PGC 119230, PGC 1000714, PGC 1436754, PGC 1694462, PGC 2822840, PGC 2826829, PGC 3098124
Loaded 88 / 88 constellation records successfully for culture "modern"
Loaded 85 / 85 constellation art records successfully for culture "modern"
Loaded 88 / 88 constellation names
Loading constellation boundary data ... 
Loaded 782 constellation boundary segments
Loaded 84 / 84 asterism records successfully for culture "modern"
Loaded 74 / 74 asterism names
Initializing basic GL shaders... 
Creating GUI ...
Loading Style file ":/graphicGui/normalStyle.css"
Loaded plugin "EquationOfTime"
Loaded plugin "Exoplanets"
[Exoplanets] Version of the format of the catalog: 1
[Exoplanets] loading catalog file: "C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\modules\\Exoplanets\\exoplanets.json"
Loaded plugin "MeteorShowers"
[MeteorShowersMgr] Loading catalog file: "C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\modules\\MeteorShowers\\MeteorShowers.json"
[MeteorShowersMgr] Version of the format of the catalog: 2
Loaded plugin "Novae"
[Novae] version of the catalog: 1
[Novae] loading catalog file: "C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\modules\\Novae\\novae.json"
Loaded plugin "Oculars"
Oculars::validateAndLoadIniFile() ocular.ini exists at:  "C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\modules\\Oculars\\ocular.ini" . Checking version...
Oculars::validateAndLoadIniFile() found existing ini file version  3.1
Loaded plugin "Satellites"
setFlagUmbraAtFixedDistance() currently not possible
[Satellites] loading catalogue file: "C:\\Users\\balto\\AppData\\Roaming\\Stellarium\\modules\\Satellites\\satellites.json"
Satellite has invalid orbit: "ERBS" "15354"
Satellite has invalid orbit: "FALCONSAT-3" "30776"
Satellite has invalid orbit: "COSMO-SKYMED 3" "33412"
Satellite has invalid orbit: "TET-1" "38710"
Satellite has invalid orbit: "CZ-2C R/B" "40306"
Satellite has invalid orbit: "FLOCK 3P-4" "41965"
Satellite has invalid orbit: "FLOCK 3P-58" "41977"
Satellite has invalid orbit: "FLOCK 3P-57" "41978"
Satellite has invalid orbit: "LEMUR-2-JENNYBARNA" "42753"
Satellite has invalid orbit: "FLOCK 2K-40" "42865"
Satellite has invalid orbit: "FLOCK 2K-39" "42866"
Satellite has invalid orbit: "FLOCK 2K-32" "42870"
Satellite has invalid orbit: "FLOCK 2K-19" "42886"
Satellite has invalid orbit: "FLOCK 2K-15" "42890"
Satellite has invalid orbit: "SPACEBEE-3" "43140"
Satellite has invalid orbit: "SURFSAT" "43614"
Satellite has invalid orbit: "ELFIN-B" "43616"
Satellite has invalid orbit: "FLOCK 3R-9" "43723"
Satellite has invalid orbit: "FLOCK 3R-6" "43741"
Satellite has invalid orbit: "LEMUR-2-VLADIMIR" "43746"
Satellite has invalid orbit: "ISS DEB (SEDA-AP)" "43870"
Satellite has invalid orbit: "FLOCK 3K-4" "43893"
Satellite has invalid orbit: "FLOCK 3K-10" "43905"
Satellite has invalid orbit: "MYSat 1" "44045"
Satellite has invalid orbit: "STARLINK-24" "44238"
Satellite has invalid orbit: "STARLINK-71" "44252"
Satellite has invalid orbit: "OCULUS-ASR" "44348"
Satellite has invalid orbit: "PSAT2 (NO-104)" "44354"
Satellite has invalid orbit: "PAINANI 1" "44365"
Satellite has invalid orbit: "STARLINK-1099" "44918"
Satellite has invalid orbit: "STPSAT-4" "45043"
Satellite has invalid orbit: "SORTIE" "45264"
Satellite has invalid orbit: "STARLINK-1322" "45552"
Satellite has invalid orbit: "STARLINK-1469" "45752"
Satellite has invalid orbit: "RED-EYE 2 (MERLOT)" "45800"
Satellite has invalid orbit: "RED-EYE 3 (CABERNET)" "45809"
Satellite has invalid orbit: "STARLINK-1597" "46142"
Satellite has invalid orbit: "STARLINK-1823" "46720"
Satellite has invalid orbit: "STARLINK-1885" "47168"
Satellite has invalid orbit: "STARLINK-1952" "47349"
Satellite has invalid orbit: "STARLINK-2306" "47988"
Satellite has invalid orbit: "STARLINK-2704" "48643"
Satellite has invalid orbit: "ISS DEB" "48833"
Satellite has invalid orbit: "RAMSAT" "48850"
Satellite has invalid orbit: "BD-28" "48867"
Satellite has invalid orbit: "ONEWEB-0321" "49191"
Satellite has invalid orbit: "ONEWEB-0338" "49204"
Satellite has invalid orbit: "ONEWEB-0357" "49219"
Satellite has invalid orbit: "BINAR-1" "49272"
Satellite has invalid orbit: "CUAVA-1" "49275"
Satellite has invalid orbit: "ONEWEB-0420" "50495"
Satellite has invalid orbit: "1998-067TD" "51441"
Satellite has invalid orbit: "PATCOOL" "51442"
Satellite has invalid orbit: "LIGHT-1" "51509"
Satellite has invalid orbit: "STARLINK-3688" "52310"
Satellite has invalid orbit: "STARLINK-3841" "52369"
Satellite has invalid orbit: "STARLINK-3762" "52374"
Satellite has invalid orbit: "STARLINK-3800" "52379"
Satellite has invalid orbit: "STARLINK-3859" "52470"
Satellite has invalid orbit: "STARLINK-3839" "52480"
Satellite has invalid orbit: "STARLINK-3875" "52483"
Satellite has invalid orbit: "STARLINK-3935" "52549"
Satellite has invalid orbit: "STARLINK-3911" "52565"
Satellite has invalid orbit: "STARLINK-3924" "52570"
Satellite has invalid orbit: "STARLINK-3917" "52588"
Satellite has invalid orbit: "STARLINK-3921" "52613"
Satellite has invalid orbit: "STARLINK-3982" "52614"
Satellite has invalid orbit: "STARLINK-4015" "52616"
Satellite has invalid orbit: "STARLINK-3996" "52617"
Satellite has invalid orbit: "STARLINK-3983" "52618"
Satellite has invalid orbit: "STARLINK-3984" "52621"
Satellite has invalid orbit: "STARLINK-3989" "52630"
Satellite has invalid orbit: "STARLINK-3995" "52632"
Satellite has invalid orbit: "STARLINK-3974" "52634"
Satellite has invalid orbit: "STARLINK-3978" "52635"
Satellite has invalid orbit: "STARLINK-3967" "52636"
Satellite has invalid orbit: "STARLINK-3971" "52642"
Satellite has invalid orbit: "STARLINK-3985" "52644"
Satellite has invalid orbit: "STARLINK-3973" "52645"
Satellite has invalid orbit: "STARLINK-3832" "52646"
Satellite has invalid orbit: "STARLINK-3846" "52647"
Satellite has invalid orbit: "STARLINK-4028" "52677"
Satellite has invalid orbit: "STARLINK-4074" "52678"
Satellite has invalid orbit: "STARLINK-4068" "52681"
Satellite has invalid orbit: "STARLINK-4039" "52683"
Satellite has invalid orbit: "STARLINK-4072" "52694"
Satellite has invalid orbit: "STARLINK-4041" "52695"
Satellite has invalid orbit: "STARLINK-4025" "52696"
Satellite has invalid orbit: "STARLINK-3999" "52704"
Loaded plugin "SolarSystemEditor"
Using the ssystem_minor.ini file that already exists in the user directory...
Loaded plugin "TelescopeControl"
[TelescopeControl] Only embedded telescope servers are available.
[TelescopeControl] Adding device model: "Meade AutoStar compatible" "Any telescope or telescope mount compatible with Meade's AutoStar controller." "TelescopeServerLx200" 500000
[TelescopeControl] Adding device model: "Meade LX200 (compatible)" "Any telescope or telescope mount compatible with Meade LX200." "TelescopeServerLx200" 500000
[TelescopeControl] Adding device model: "Meade ETX70 (#494 Autostar, #506 CCS)" "Meade's ETX70 with the #494 Autostar controller and the #506 Connector Cable Set." "TelescopeServerLx200" 1500000
[TelescopeControl] Adding device model: "Losmandy G-11" "Losmandy's G-11 telescope mount." "TelescopeServerLx200" 500000
[TelescopeControl] Adding device model: "Wildcard Innovations Argo Navis (Meade mode)" "Wildcard Innovations' Argo Navis DTC in Meade LX200 emulation mode." "TelescopeServerLx200" 500000
[TelescopeControl] Adding device model: "Celestron NexStar (compatible)" "Any telescope or telescope mount compatible with Celestron NexStar." "TelescopeServerNexStar" 500000
[TelescopeControl] Adding device model: "Sky-Watcher SynScan (version 3 or later)" "Any Sky-Watcher mount that uses version 3 or later of the SynScan hand controller." "TelescopeServerNexStar" 500000
[TelescopeControl] Adding device model: "Sky-Watcher SynScan AZ GOTO" "The Sky-Watcher SynScan AZ GOTO mount used in a number of telescope models." "TelescopeServerNexStar" 500000
[TelescopeControl] Loaded successfully 3 telescopes.
Initializing planets GL shaders... 
  OpenType support missing for "Verdana", script 11
  OpenType support missing for "Arial", script 11
  OpenType support missing for "MS UI Gothic", script 11
  OpenType support missing for "SimSun", script 11
  OpenType support missing for "Segoe UI Emoji", script 11
  OpenType support missing for "Segoe UI Symbol", script 11

**Comment by barbu34** [comment]
And same issue if i zoom the screen. The galaxies and globular clusters change of color.
ZoomFOV-70,9
ZoomFOV-83,7

**Comment by barbu34** [comment]
Just notice that there is no probleme whith open clusters nebula and planetary nebula.

**Pull request merged**